Add output visualization functions

maltelueken opened the pull request:
Adds plotting functions:

- Plot population pyramid
- Plot life expectancy
- Plot time series

They imitate the plots from the old Java GUI.

Adds regression tests against plot snapshots.

Adds a function to load the simulation results.

Adds a function to download DYNAMO-HIA Java app from GitHub releases.

Completes the example vignette.

f-hafner commented:
for my review, important: check the Rmd file. fix the problem with github: need env variable GITHUB_PAT.

maltelueken commented:
I made the GITHUB_PAT optional now, so it should work without setting the environment variable.

f-hafner commented on vignettes/hiaR.Rmd:
```diff
@@ -150,30 +191,121 @@ create_simulation_dir(
+  repo_url = "https://github.com/Dynamo-HIA/dynamo-hia",
+  release_tag = "v3.0.0-beta.1",
+  dest_dir = temp_dir,
+  github_pat = Sys.getenv("GITHUB_PAT")
```
I had to remove the last argument to make the function call work.

f-hafner commented on R/utils.R:
```diff
@@ -365,3 +365,173 @@ filter_df_from_server_data <- function(
+#' }
+#'
+download_github_release <- function(repo_url = "https://github.com/Dynamo-HIA/dynamo-hia",
+                                    release_tag = "v3.0.0-beta.1",
```
If possible, I would try to avoid hard-coding the release name here. depending on how frequently we expect new releases of HIA, we could (i) mention this explicitly somewhere in the developer documentation and/or (ii) store the variable explicitly somewhere in a "globals" file (as an env variable), or alternatively (iii) automatically get the name of the latest release from github?

maltelueken replied:
Hmm, if we require an env variable, this makes it more complicated for the user (especially nontechnical ones). The problem with using the latest release is that the function with the default values is not reproducible anymore, i.e., it might give different results at different times after a new release was made. I will mention it in the developer documentation.

f-hafner replied:
ok!

f-hafner commented on vignettes/hiaR.Rmd:
````diff
@@ -150,30 +191,121 @@ create_simulation_dir(
 })
 ```
-The function returns `TRUE` indicating that the simulation was successful. The results of the simulation are stored in the folder of the simulation.
 
````
my `knit` call breaks here with this message:
```
Warning: setting stdout = TRUEError in run_dynamo_hia(batch_file_name, fs::path(temp_dir, Sys.getenv("DYNAMO_HIA_PATH"))) : 
  2025-01-06 10:54:55,745 DEBUG [nl.rivm.emi.dynamo.global.CommandLineSchemaFileProviderImpl] - <Instantiating : CommandLineSchemaFileProviderImpl>2025-01-06 10:54:55,745 DEBUG [nl.rivm.emi.dynamo.global.CommandLineSchemaFileProviderImpl] - <Schema location : /tmp/Rtmpcc08fA/file3e4a34d25ebaa/schemas/>Started, arguments: simulation_names.txt, 2025-01-06 10:54:55,776 DEBUG [org.apache.commons.configuration.ConfigurationUtils] - <ConfigurationUtils.locate(): base is null, name is /tmp/Rtmpcc08fA/file3e4a34d25ebaa/Simulations/Netherlands_Smoking_Intervention/configuration.xml>2025-01-06 10:54:55,776 DEBUG [org.apache.commons.configuration.ConfigurationUtils] - <Loading configuration from the absolute path /tmp/Rtmpcc08fA/file3e4a34d25ebaa/Simulations/Netherlands_Smoking_Intervention/configuration.xml>2025-01-06 10:54:55,779 DEBUG [org.apache.commons.configuration.XMLConfigurationToo] - <getting document builder for org.apache.commons.configuration.HierarchicalConfiguration$Node@258
```

maltelueken replied:
I tried to fix this but I don't think it worked. I think the issue lies within DYNAMO-HIA itself because it runs on Windows without issues, but I will investigate.

f-hafner reviewed (changes requested):
See my comments; I cannot knit. If it's easier, we can debug together and I look at the rest of the vignette when it works

maltelueken commented:
Addressed your points and fixed the tests. The MacOS action is sometimes failing because of API access limiting, but it has passed for previous commits.

f-hafner commented on README.Rmd:
```diff
@@ -98,6 +98,10 @@ For R Shiny modules, use the following naming pattern:
 Any functions from `shiny` and `shinyFiles` can be used directly without the `shiny::` prefix.
 For any other packages, be explicit and use `mypackage::myfunction()`.
 
+#### Before Making a Release
```
this refers to a release of the HIA program, not the hiaR package, right? -> I would make this specific about for which software a release is made.
